File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
def solution(word):
    w = ['A','E','I','O','U']
    answer = 0
    count = 0
    def dfs(current):
        nonlocal answer, count
        if current == word:
            answer = count
            return
        
        if len(current) == 5:
            return
        
        for i in w:
            count += 1
            dfs(current+i)
            
    dfs("")
    
    return answer

File: 프로그래머스/2/84512. 모음 사전/모음 사전.cpp
#include <string>
#include <vector>
#include <functional>

using namespace std;

int solution(string word) {
    string w = "AEIOU";
    int answer = 0;
    int count = 0;

    function<void(string)> dfs = [&](string current) {
        if (current == word) {
            answer = count;
            return;
        }

        if (current.length() == 5) {
            return;
        }

        for (char c : w) {
            count++;
            dfs(current + c);
        }
    };

    dfs("");
    
    return answer;
}